async-nats/src/jetstream/consumer/push.rs

```diff
@@ -10,19 +10,68 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::time::Duration;
+use super::{AckPolicy, Consumer, DeliverPolicy, FromConsumer, IntoConsumerConfig, ReplayPolicy};
+use crate::{
+    jetstream::{self, Context, Message},
+    Error, StatusCode, Subscriber,
+};
 
+use bytes::Bytes;
 use serde::{Deserialize, Serialize};
+use std::pin::Pin;
+use std::task::{self, Poll};
+use std::time::Duration;
 
-use crate::{jetstream, Error};
+impl Consumer<Config> {
+    pub async fn stream(&self) -> Result<Stream, Error> {
+        let deliver_subject = self.info.config.deliver_subject.clone().unwrap();
+        let subscriber = self.context.client.subscribe(deliver_subject).await?;
 
-use super::{AckPolicy, Consumer, DeliverPolicy, FromConsumer, IntoConsumerConfig, ReplayPolicy};
+        Ok(Stream {
+            context: self.context.clone(),
+            subscriber,
+        })
+    }
+}
 
-impl Consumer<Config> {
-    pub fn push(&self) {
-        println!("push");
+pub struct Stream {
+    context: Context,
+    subscriber: Subscriber,
+}
+
+impl futures::Stream for Stream {
+    type Item = Message;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut task::Context<'_>) -> Poll<Option<Self::Item>> {
+        match self.subscriber.receiver.poll_recv(cx) {
+            Poll::Ready(maybe_message) => match maybe_message {
+                Some(message) => match message.status {
+                    Some(StatusCode::IDLE_HEARBEAT) => {
+                        if let Some(subject) = message.reply {
+                            let client = self.context.client.clone();
+                            tokio::task::spawn(async move {
+                                client
+                                    .publish(subject, Bytes::from_static(b""))
+                                    .await
+                                    .unwrap();
+                            });
+                        }
+
+                        Poll::Pending
+                    }
+                    Some(_) => Poll::Pending,
```

Comment:
this will deadlock if we got a message we didn't expect and we didn't poll subscriber again.
Same case as in Pull Consumer.

Reply:
Doesn't seem to be necessary to eagerly poll for channels? at-least I can't make it deadlock in tests.

Reply:
Nevermind, you were right, bad tests, fixed 😁